src/nbl/asset/utils/ISPIRVDebloater.cpp

```diff
@@ -0,0 +1,115 @@
+#include "nbl/asset/utils/ISPIRVDebloater.h"
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl_spirv_cross/spirv.hpp"
+
+#include "nbl/core/declarations.h"
+#include "nbl/system/ILogger.h"
+
+
+using namespace nbl::asset;
+
+ISPIRVDebloater::ISPIRVDebloater()
+{
+    constexpr auto optimizationPasses = std::array{
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_VARIABLES,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_CONSTANTS,
+        // TODO: remove unused type. no dedicated flag for type. maybe need agressive dce to remove type.
```

Comment:
DeadMember optimization may remove the members from structs, and then you have empty structs so maybe empty `OpStruct` will get removed?

Then the only things left behind would be `OpType` for SPIR-V types, I wonder what pass can remove those...